server/src/main/java/io/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -91,7 +92,7 @@ public Void withHandle(Handle handle) throws Exception
                     getSupervisorsTable()
                 )
             )
-                  .bind("id", String.format("%s_%s", id, version))
+                  .bind("id", String.format("%s_%s_%s", id, version, UUIDUtils.generateUuid()))
```

Comment:
yes, this is a good point, it needs to increment. Maybe `version` should be an auto incrementing field? Time isn't really guaranteed to proceed linearly anyway, so that'd be more reliable than using the current timestamp. And it would fix this bug too.